presto-docs/src/main/sphinx/sql/select.rst

```diff
@@ -92,6 +92,62 @@ Additionally, the relations within a ``WITH`` clause can chain::
     relation is used. This means that if the relation is used more than once and the query
     is non-deterministic, the results may be different each time.
 
+
+WITH RECURSIVE Clause
+---------------------
+
+The ``WITH RECURSIVE`` clause is a variant of the ``WITH`` clause. It defines a
+``UNION`` of two relations. The first relation is called the *recursion base*,
+and the second relation is called the *recursion step*:
+
+.. code-block:: none
+
+    WITH RECURSIVE T(n) AS (
+    <recursion base relation>
+    UNION (ALL | DISTINCT)
+    <recursion step relation: SELECT .. FROM ... T...>
+    )
+    ... SELECT ... FROM T
+
+Presto supports recursive ``WITH``-queries with a single recursive reference to
+a ``WITH``-query from within the query. The name ``T`` of the query ``T`` can be
+mentioned once in the ``FROM`` clause of the recursion step relation.
+
+Logically, the recursive query processing is a sequence of actions:
+
+- compute the recursion base
+- compute the recursion step using the precomputed base where the relation name
+  is mentioned
+- compute the next step using the previous step
+- repeat until the recursion converges, which means an empty relation is
+  obtained
+- union the base and all the steps, and return it as the result
+
+The types of the returned columns are those of the base relation. Therefore it
+is required that types in the step relation can be coerced to base relation
+types.
+
+The ``RECURSIVE`` clause applies to all queries in the ``WITH`` list, but not
+all of them must be recursive. If a ``WITH``-query is not shaped according to
+the rules mentioned above or it does not contain a recursive reference, it is
+processed like a regular ``WITH``-query. Column aliases are mandatory for all
+the queries in the recursive ``WITH`` list.
+
+The following limitations apply as a result of following the SQL standard and
+due to implementation choices, in addition to ``WITH`` clause limitations:
+
+- only single-element recursive cycles are supported. Like in regular
+  ``WITH``-queries, references to previous queries in the ``WITH`` list are
+  allowed. References to following queries are forbidden.
+- usage of outer joins, set operations, limit clause, and others is not always
+  allowed in the step relation
+- recursion depth is fixed, defaults to ``1``, and doesn't depend on the actual
+  query results
```

Comment:
It defaults to `10` if I recall correctly.

Reply:
fixed